boilerplate/app/components/Toggle/Checkbox.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react"
+import { Image, ImageStyle, StyleProp, View, ViewStyle } from "react-native"
+import Animated, { useAnimatedStyle, withTiming } from "react-native-reanimated"
+import { colors } from "../../theme"
+import { iconRegistry, IconTypes } from "../Icon"
+import { $inputOuterBase, BaseToggleInputProps, ToggleProps, Toggle } from "./Toggle"
+
+export interface CheckboxToggleProps extends Omit<ToggleProps<CheckboxInputProps>, "ToggleInput"> {
+  /**
+   * Optional style prop that affects the Image component.
+   */
+  inputDetailStyle?: ImageStyle
+  /**
+   * Checkbox-only prop that changes the icon used for the "on" state.
+   */
+  checkboxIcon?: IconTypes
```

Comment:
Should we change this to just be `icon` since it belongs to the `Checkbox` component?

Reply:
agreed!